debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -115,6 +115,12 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
             // otherwise we might skip an incomplete transaction after restart
             boolean shouldIncreaseFromLsn = offsetContext.isSnapshotCompleted();
             while (context.isRunning()) {
+                // When reading from read-only Always On replica the default and only transaction isolation
+                // is snapshot. This means that CDC metadata are not visible for long-running transactions.
+                // It is thus necessary to restart the transaction before every read.
+                if (connectorConfig.isReadOnlyDatabaseConnection()) {
+                    dataConnection.commit();
```

Comment:
I guess I'm still not quite getting the purpose of commit if it's a read-only connection. Does the commit trigger any state change on the replica or not?